rules/ast/ast.go

```diff
@@ -371,10 +383,15 @@ func (node *ScalarFunctionCall) Eval(timestamp clientmodel.Timestamp) clientmode
 	return node.function.callFn(timestamp, node.args).(clientmodel.SampleValue)
 }
 
-func (node *VectorAggregation) labelsToGroupingKey(labels clientmodel.Metric) uint64 {
+// hashForLabels returns a hash value taken from the values of the
+// specified labels in the metric. The hash is not based on the label names.
+func hashForLabels(metric clientmodel.Metric, labels clientmodel.LabelNames) uint64 {
+	// Sort labels to guarantee the same hash across numerous calls.
```

Comment:
Btw., the client library has a pretty genius way to avoid this sorting in its internal signature algorithm: just hash each label separately and XOR the results all together. When you XOR stuff, the sort order doesn't matter.

https://github.com/prometheus/client_golang/blob/master/model/signature.go#L57-L76

We were thinking of switching Fingerprints to that algorithm as well, so maybe it would be good here too?